### ISSUE TITLE
[Bug] Capital CSS variables do not work in object style

### ISSUE THREAD

**Issue by siriwatknp** [styled-components/styled-components#3810]
## Environment

`npx envinfo --system --binaries --npmPackages styled-components,babel-plugin-styled-components --markdown --clipboard`

```md
"styled-components": "5.3.5"
```

## Reproduction

https://codesandbox.io/s/styled-components-base-forked-q77k2v?file=/index.js

## Steps to reproduce

1. Go to the official site: https://styled-components.com/docs/advanced#style-objects
2. Use this snippet in the playground
   ```jsx
    const PropsBox = styled.div(props => ({
      background: props.background,
      height: '50px',
      width: '50px',
      '--Grid-columns': '12',
    }));
    
    const PropsBox2 = styled.div`
      --Grid-columns: 12;
      width: 50px;
      height: 50px;
      background: tomato;
    `;
    
    render(
      <div>
        <PropsBox background="blue" />
        <PropsBox2 />
      </div>
    );
   ```
3. You should see this
  <img width="827" alt="Screen Shot 2565-09-13 at 10 02 36" src="https://user-images.githubusercontent.com/18292247/189798248-3b389b00-ab2a-4ced-8d6b-f93e793d4eaf.png">
4. inspect the DOM and you will see that the blue box produces an unexpected result but the tomato box works fine.
  <img width="204" alt="Screen Shot 2565-09-13 at 10 03 55" src="https://user-images.githubusercontent.com/18292247/189798417-d3be0d18-b46d-4917-94e4-622cfd6861bf.png">

## Expected Behavior

The CSS variables should be changed.

## Actual Behavior

The CSS variables are prefixed (I guess).
```js
// input
{
  '--Grid-columns': '12',
}

// output
---grid-columns: 12;
```

**Comment by agriffis** [comment]
@siriwatknp The sandbox link you provided is just the base sandbox with no changes. Could you make a sandbox that demonstrates the issue?

**Comment by siriwatknp** [comment]
> @siriwatknp The sandbox link you provided is just the base sandbox with no changes. Could you make a sandbox that demonstrates the issue?

Opps, sorry. Updated the CodeSandbox > https://codesandbox.io/s/styled-components-base-forked-q77k2v?file=/index.js

**Comment by siriwatknp** [comment]
I think I can guess what's wrong here. It seems that styled-components think a capital letter is a CSS prefix, so it compile to `-` with a lower case.

**Comment by nupur-jeswani** [comment]
Can you please assign this issue to me?

**Comment by agriffis** [comment]
@nupur-jeswani This issue is already fixed by #3812, it's just waiting to be merged.
